yamux/src/stream.rs

```diff
@@ -480,6 +480,26 @@ impl AsyncWrite for StreamHandle {
     }
 }
 
+impl Drop for StreamHandle {
+    fn drop(&mut self) {
+        if !self.event_sender.is_closed()
+            && (self.state != StreamState::Closed || self.state != StreamState::LocalClosing)
```

Comment:
Replace `||` with `&&`?

Reply:
state is only one value, how can it be equal to two states at the same time

Reply:
This is a signal that it lacks the test coverage

Reply:
state is only one value, how can it be equal to two states at the same time

Reply:
I don’t think it needs to be tested here

Reply:
As you said, x can not be a and b at at the same times, `x != a || x != b` is always true.

Reply:
you are right